src/main/scala/Plugin.scala

```diff
@@ -1,460 +0,0 @@
-            } else {
-              println(s"Missing bintray credentials $credsFile. Some bintray features depend on this.")
-              None
-            }
-          case creds => creds
-        })
-
-  def bintrayPublishSettings: Seq[Setting[_]] = bintrayCommonSettings ++ Seq(
-    credentialsFile in bintray in Global := Path.userHome / ".bintray" / ".credentials",
-    name in bintray := moduleName.value,
-    bintrayOrganization in bintray in Global := { if (sbtPlugin.value) Some("sbt") else None },
-    // todo: don't force this to be sbt-plugin-releases
-    repository in bintray in Global := { if (sbtPlugin.value) "sbt-plugin-releases" else "maven" },
-    publishMavenStyle := {
-      if (sbtPlugin.value) false else publishMavenStyle.value
-    },
-    vcsUrl in bintray := resolveVcsUrl.recover {
-      case NonFatal(e) =>
-        None
-    }.get,
-    packageLabels in bintray := Nil,
-    description in bintray <<= description,
-    // note: publishTo may not have dependencies. therefore, we can not rely well on inline overrides
-    // for inline credentials resolution we recommend defining bintrayCredentials _before_ mixing in the defaults
-    // perhaps we should try overriding something in the publishConfig setting -- https://github.com/sbt/sbt-pgp/blob/master/pgp-plugin/src/main/scala/com/typesafe/sbt/pgp/PgpSettings.scala#L124-L131
-    publishTo in bintray <<= publishToBintray,
-    resolvers in bintray <<= appendBintrayResolver,
-    credentials in bintray <<= (credentialsFile in bintray).map {
-      Credentials(_) :: Nil
-    },
-    packageAttributes in bintray <<= (sbtPlugin, sbtVersion) {
-      (plugin, sbtVersion) =>
-        if (plugin) Map(AttrNames.sbtPlugin -> Seq(Attr.Boolean(plugin)))
-        else Map.empty
-    },
-    versionAttributes in bintray <<= (crossScalaVersions, sbtPlugin, sbtVersion) {
-      (scalaVersions, plugin, sbtVersion) =>
-        val sv = Map(AttrNames.scalas -> scalaVersions.map(Attr.Version(_)))
-        if (plugin) sv ++ Map(AttrNames.sbtVersion-> Seq(Attr.Version(sbtVersion))) else sv
-    },
-    omitLicense in bintray in Global := { if (sbtPlugin.value) sbtPlugin.value else false },
-    ensureLicenses <<= ensureLicensesTask,
-    ensureCredentials <<= ensureCredentialsTask,
-    ensureBintrayPackageExists <<= ensurePackageTask,
-    publishVersionAttributes <<= publishVersionAttributesTask,
-    unpublish in bintray <<= unpublishTask.dependsOn(ensureBintrayPackageExists, ensureLicenses),
-    remoteSign in bintray <<= remoteSignTask,
-    syncMavenCentral in bintray <<= syncMavenCentralTask
-  ) ++ Seq(
-    resolvers <++= resolvers in bintray,
-    credentials <++= credentials in bintray,
-    publishTo <<= publishTo in bintray,
-    // We attach this to publish configruation, so that publish-signed in pgp plugin can work.
-    publishConfiguration <<= publishConfiguration.dependsOn(ensureBintrayPackageExists, ensureLicenses), 
```

Comment:
> I'm really happy with the overall set of changes but I kind of wish we did this is smaller iterations :/

FWIW I did split this PR into series of incremental commits and the points you raised were done separately:
- 822d3cdac98d1d97cce723c8641ffe22a9d33384
- a01e701970fffc348f4ddcae84bfa2e7ac8373cf

I agree with the idea of "defaults should *just work*." These changes were made because it didn't just work for sbt-web (sbt/sbt-web#106) and others.